pkg/odo/cli/telemetry/telemetry.go

```diff
@@ -0,0 +1,80 @@
+package telemetry
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/preference"
+	"github.com/openshift/odo/pkg/segment"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/klog"
+)
+
+const RecommendedCommandName = "telemetry"
+
+type TelemetryOptions struct {
+	telemetryData segment.TelemetryData
+}
+
+func NewTelemetryOptions() *TelemetryOptions {
+	return &TelemetryOptions{}
+}
+
+func (o *TelemetryOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	err = json.Unmarshal([]byte(args[0]), &o.telemetryData)
```

Comment:
This will panic if the caller doesn't provide any args. We can add `Args:cobra.ExactArgs(1),` to telemetryCmd for preventing this panic.